### PR TITLE
add stationxml decoder elements

### PR DESCRIPTION
**Pull request by ozym**
This PR follows on from the encoder work.

Given a byte representation of an XML file, the decode will, depending on required version, extract the appropriate parts and return a cleaned byte slice. This code is primarily used to compare two different StationXML files and attempts to make the options uniform allowing comparisons.

### CODE REVIEW

File: internal/stationxml/decoder10.go [hunk 1]
```diff
@@ -0,0 +1,171 @@
...
+					stage.Coefficients.ResourceId = ""
+				}
+				for v := range cha.Response.Stage {
+					stage := &cha.Response.Stage[v]
+					if stage.Decimation == nil {
+						continue
+					}
+					stage.Decimation.Delay.Value = 0.0
+					stage.Decimation.Correction.Value = 0.0
+				}
```
Comment (bpeng) [inline internal/stationxml/decoder10.go:138]: the above 5 loops can be combined into 1, use something like
```
if stage.Polynomial != nil {
	stage.Polynomial.Name = ""
	stage.Polynomial.ResourceId = ""
}
...
```

File: internal/stationxml/decoder10.go [hunk 2]
```diff
@@ -0,0 +1,171 @@
...
+			}
+
+			for i := range sta.Comment {
+				sta.Comment[i].Id = stationxml.CounterType(i + 1)
+			}
+
+			net.StartDate = stationxml.DateTime{}
+			net.EndDate = stationxml.DateTime{}
+		}
```
Comment (bpeng) [inline internal/stationxml/decoder10.go:150]: channels are not sorted as version11&12, is this intentional?

**Review by bpeng — CHANGES_REQUESTED** [review]
code looks good, small suggestions
separate structs for 3 versions have duplicate code, not sure if it is feasible to use one struct

**Comment by ozym** [comment]
The duplicate code is related to importing different modules. Will update other comments.

**Comment by ozym** [comment]
@bpeng have updated code.

**Review by bpeng — APPROVED** [review]
OK

**Pull request merged**